dynamiq/memory/backend/in_memory.py

```diff
@@ -8,6 +11,46 @@ class InMemoryError(Exception):
     pass
 
 
+class BM25:
+    """BM25 implementation for scoring documents."""
+
+    def __init__(self, messages, k1=1.5, b=0.75):
+        """Initialize with a list of messages and parameters for BM25."""
```

Comment:
Let's keep the class implementation in a more general format. Instead of depending on the message format, just use a list of texts.